src/main/java/com/google/api/codegen/transformer/SurfaceTransformerContext.java

```diff
@@ -149,6 +149,21 @@ public MethodTransformerContext asMethodContext(Method method) {
     return methods;
   }
 
+  /** Returns a list of methods with samples, similar to getSupportedMethods,
+   *  but also filter out private methods.
+   */
+  public List<Method> getMethodsWithSamples() {
```

Comment:
It seems like "getPublicMethods" would be better - there is no check in the logic for having samples.